life_expectancy/cleaning.py

```diff
@@ -0,0 +1,90 @@
+"""
+Module that contains all functions to clean dataset.
+"""
+
+import argparse
+import os
+from pathlib import Path
+
+import pandas as pd
+
+CURRENT_DIR = Path(__file__).parent.resolve()
+DATA_DIR = os.path.join(CURRENT_DIR, "data")
+FILE_PATH = os.path.join(DATA_DIR, "eu_life_expectancy_raw.tsv")
+FILE_NAME = "pt_life_expectancy.csv"
+
+
+def load_data(file_path: str) -> pd.DataFrame:
+    """
+    Load the raw data from tsv file.
+
+    :param file_path: The file path.
+    :returns: A pandas dataframe with the raw data.
+    """
+    dataset = pd.read_csv(file_path, sep="\t")
+
+    return dataset
+
+
+def save_data(dataset: pd.DataFrame, file_name: str, data_dir_path: str) -> None:
+    """
+    Save the data into a file.
+
+    :param dataset: A pandas dataframe with the data to save to a file.
+    :param file_name: The name of the file to save.
+    :param data_dir_path: The path of data directory.
+    """
+    dataset.to_csv(os.path.join(data_dir_path, file_name), index=False)
+
+
+def clean_data(dataset: pd.DataFrame, country: str = "PT") -> pd.DataFrame:
+    """
+    Clean the raw data.
+
+    :param dataset: A pandas dataframe with the data to clean.
+    :param country: The name of country.
+    :returns: A pandas dataframe with the cleaned data.
+    """
+    dataset[["unit", "sex", "age", "region"]] = dataset[
+        "unit,sex,age,geo\\time"
+    ].str.split(",", expand=True)
+
+    dataset.drop(columns=["unit,sex,age,geo\\time"], inplace=True)
+
+    dataset = pd.melt(
+        dataset, id_vars=["unit", "sex", "age", "region"], var_name="year"
+    )
+
+    dataset["year"] = dataset["year"].astype("int")
+    dataset["value"] = dataset.value.str.extract(r"(\d+\.\d)").astype("float")
+
+    clean_dataset = dataset.dropna()
+
+    dataset_output = clean_dataset[clean_dataset.region == country]
```

Comment:
Do you really ned to create other variables here (clean_dataset and dataset_output)?

Reply:
in my opinion, it's more clear if I define those variables. Then when I read the code i know that function clean dataset and return a dataset for a specific region. @MrCordeiro what is your opinion about that? :)